### PR TITLE
Added ROLAND

### PR DESCRIPTION
**Pull request by benjaminnNgo**
### Summary / Description

Added ROLAND for linkproppred

**Related Issues:** #344 

#### Type of Change

- [ ] Bug fix
- [x] New feature
- [ ] Breaking Change
- [ ] Refactoring
- [x] Documentation update

#### Test Evidence

Describe how this PR has been tested.

- [x] Unit tests
- [x] Integration tests
- [ ] Performance tests

#### Questions / Discussion Points
N/A

### CODE REVIEW

File: examples/linkproppred/roland.py [hunk 1]
```diff
@@ -0,0 +1,322 @@
+import argparse
+
+# ==
```
Comment (Jacob-Chmura) [inline examples/linkproppred/roland.py]:
```suggestion
```

File: examples/linkproppred/roland.py [hunk 2]
```diff
@@ -0,0 +1,322 @@
...
+        input_channel: int,
+        nhid: int,
+        num_nodes: int,
+        dropout: float = 0.0,
+        update: str = 'learnable',
+        tau: float = 0.5,
+    ) -> None:
+        super().__init__()
+        self.recurrent = ROLAND(input_channel, nhid, num_nodes, dropout, update, tau)
+        self.linear = nn.Linear(nhid, nhid)
```
Comment (Jacob-Chmura) [inline examples/linkproppred/roland.py]: Is this being used?

File: examples/linkproppred/roland.py [hunk 3]
```diff
@@ -0,0 +1,322 @@
...
+                z = encoder(
+                    snapshot_batch,
+                    static_node_feats,
+                    last_embeddings,
+                    num_previous_edges=prev_num_edge,
+                    num_current_edges=curr_num_edge,
+                )
+                last_embeddings = z
+                for i in range(len(z)):
+                    z[i] = z[i].detach()
```
Comment (Jacob-Chmura) [inline examples/linkproppred/roland.py]: Is this a loop over all node embeddings? Could we `detach` in one call?

File: examples/linkproppred/roland.py [hunk 4]
```diff
@@ -0,0 +1,322 @@
...
+
+for epoch in range(1, args.epochs + 1):
+    last_embeddings = [
+        torch.Tensor(
+            [[0 for _ in range(args.embed_dim)] for _ in range(test_dg.num_nodes)]
+        ).to(args.device),
+        torch.Tensor(
+            [[0 for _ in range(args.embed_dim)] for _ in range(test_dg.num_nodes)]
+        ).to(args.device),
+    ]
```
Comment (Jacob-Chmura) [inline examples/linkproppred/roland.py]:
```suggestion
        torch.zeros(2, args.embed_dim, test_dg.num_nodes, device=args.device)
```

File: examples/linkproppred/roland.py (hunk 3 above)
Comment (ntgbaoo) [reply examples/linkproppred/roland.py]: ROLAND maintains 2 hidden states, independent actually, so z here is a list of two hidden states.
We can instead do this:
```python
z[0], z[1] = z[0].detach(), z[1].detach()
```

File: examples/linkproppred/roland.py (hunk 4 above)
Comment (ntgbaoo) [reply examples/linkproppred/roland.py]: I will do this:
```python
last_embeddings = [
        torch.zeros(test_dg.num_nodes,args.embed_dim, device=args.device),
        torch.zeros(test_dg.num_nodes,args.embed_dim, device=args.device),
    ]
```

Since ROLAND maintains 2 hidden states independently, it is quite similar to `c` and `h` in `GCLSTM`

**Review by Jacob-Chmura — APPROVED** [review]
